pkg/ipam/ipam.go

```diff
@@ -39,6 +39,28 @@ func ExecDel(plugin string, netconf []byte) error {
 	return invoke.ExecPluginWithoutResult(invoke.Find(plugin), netconf, invoke.ArgsFromEnv())
 }
 
+func addIPConfigToLink(ipConfig *types.IPConfig, link netlink.Link, ifName string) error {
+	addr := &netlink.Addr{IPNet: &ipConfig.IP, Label: ""}
+	if err := netlink.AddrAdd(link, addr); err != nil {
+		return fmt.Errorf("failed to add addr to %q: %v", ifName, err)
+	}
+
+	for _, r := range ipConfig.Routes {
+		gw := r.GW
+		if gw == nil {
+			gw = ipConfig.Gateway
+		}
+		if err := ip.AddRoute(&r.Dst, gw, link); err != nil {
+			// we skip over duplicate routes as we assume the first one wins
```

Comment:
Not my code, that was already there. See [here](https://github.com/aanm/cni/blame/0becca2e96e3cf2fac1dc383e8890d617e306fe3/pkg/ipam/ipam.go#L66).
Ping @squaremo